packages/react-components/react-positioning/src/middleware/shift.ts

```diff
@@ -14,10 +14,22 @@ export interface ShiftMiddlewareOptions
  * Wraps the floating UI shift middleware for easier usage of our options
  */
 export function shift(options: ShiftMiddlewareOptions) {
-  const { hasScrollableElement, disableTether, overflowBoundary, container, overflowBoundaryPadding, isRtl } = options;
+  const {
+    hasScrollableElement,
+    shiftToCoverTarget,
+    disableTether,
+    overflowBoundary,
+    container,
+    overflowBoundaryPadding,
+    isRtl,
+  } = options;
 
   return baseShift({
     ...(hasScrollableElement && { boundary: 'clippingAncestors' }),
+    ...(shiftToCoverTarget && {
+      crossAxis: true,
+      limiter: limitShift({ crossAxis: true, mainAxis: false }),
```

Comment:
would you be able to test what the consequences of applying `shiftToCoverTarget` are to this example? https://react.fluentui.dev/?path=/docs/components-menu-menu--docs#nested-submenus-responsiveness

Reply:
And submenus in general actually

Reply:
It works as expected, submenu falls back to above when boundary width gets smaller: 
<img width="432" alt="Screenshot 2024-12-16 at 15 13 01" src="https://github.com/user-attachments/assets/46de9de6-d924-4b1a-9406-fa06412d552c" />

And submenu can shift to cover the trigger item when the height gets smaller:
<img width="214" alt="image" src="https://github.com/user-attachments/assets/693aea01-3666-41dc-b154-5f3447456c06" />